python_tests/test_python_frontend.py

```diff
@@ -2023,6 +2023,40 @@ def fusion_func(fd: FusionDefinition) -> None:
         nvf_out, _ = self.exec_nvfuser(fusion_func, inputs)
         self.assertEqual(eager_out, nvf_out[0])
 
+    def test_graphviz(self):
+        inputs = [
+            torch.ones(2, 4, 8, device="cuda"),
+            torch.ones(2, 4, 8, device="cuda"),
+        ]
+
+        with FusionDefinition() as fd:
+            t0 = fd.from_pytorch(inputs[0])
+            t1 = fd.from_pytorch(inputs[1])
+            c0 = fd.define_constant(3.0)
+
+            t2 = fd.ops.add(t0, t1)
+            t3 = fd.ops.mul(t2, c0)
+            t4 = fd.ops.sum(t3, [-1], False, DataType.Float)
+
+            fd.add_output(t4)
+
+        # test generating a dot file before execution. This will throw an
+        # assertion error if detail_level is Explicit or Verbose
+        pre_exec = fd.to_graphviz()
+        print("pre_exec=", pre_exec)
+
+        _ = fd.execute(inputs)
+
+        # test generating a dot file after execution with different detail levels
+        co = fd.to_graphviz("computeOnly")
+        print("ComputeOnly=", co)
+        b = fd.to_graphviz("basic")
+        print("Basic=", b)
+        e = fd.to_graphviz("Explicit")
+        print("Explicit=", e)
+        v = fd.to_graphviz("VERBOSE")
+        print("Verbose=", v)
```

Comment:
These work now; I updated the gist to show the output. The schedule's don't look particularly interesting yet in that example. It could be the Fusion I've selected or the graphviz export function. I'll play with it some more, but if the graphviz function needs tweaking that may be a topic for another PR.

Reply:
By matching the other debug interfaces, I was hoping for something like:

```
fd.last_graphviz()
fd.graphviz_for(inputs, tensor_transforms=True)
```

Therefore, the functions look standardized for the convenience of the user. 

I am not sure what the strings represent but I assume they correspond to something like the unscheduled ir, scheduled ir, and the scheduled ir with tensor transforms.   If that is the case it would be good to match up the options with the other debug functions like `last_scheduled_fusion_ir()` that has an option to toggle on `tensor_transforms` so the options are consistent between the 3 interfaces when describing the same things.  If you have better option names like maybe `tensor_transforms` is not the best or if you think the 3 sets of name should be something different (but consistent), please suggest them.  This is a great time to make the interfaces better!